File: patch-onboarding.mjs
#!/usr/bin/env node

/**
 * Script pour corriger le fichier OnboardingContextNew.tsx
 * Remplace l'appel RPC create_company_with_user par une logique directe
 */

import fs from 'fs';
import path from 'path';

const filePath = 'src/contexts/OnboardingContextNew.tsx';

if (!fs.existsSync(filePath)) {
  console.error(`❌ File not found: ${filePath}`);
  process.exit(1);
}

let content = fs.readFileSync(filePath, 'utf8');

// Check if already patched
if (content.includes('createCompanyDirectly(user.id')) {
  console.log('✅ File already patched');
  process.exit(0);
}

// Add import if not present
if (!content.includes("import { createCompanyDirectly }")) {
  const importPosition = content.indexOf('import { devLogger }');
  if (importPosition > 0) {
    const endOfLine = content.indexOf('\n', importPosition);
    const importStatement = "import { createCompanyDirectly } from '@/helpers/createCompanyHelper';\n";
    content = content.slice(0, endOfLine + 1) + importStatement + content.slice(endOfLine + 1);
  }
}

// Remove the authedClient creation - we don't need it anymore
const authedClientStart = content.indexOf('const authedClient = session?.access_token');
if (authedClientStart > 0) {
  const authedClientEnd = content.indexOf(';', content.indexOf('persistSession: false', authedClientStart)) + 1;
  const authedClientLine = content.slice(authedClientStart, authedClientEnd);
  
  console.log('Removing authedClient creation...');
  content = content.replace(authedClientLine, '// authedClient no longer needed');
}

// Replace the RPC call
const oldRpcCode = `const { data: result, error: rpcError } = await authedClient.rpc('create_company_with_user'`;
if (content.includes(oldRpcCode)) {
  console.log('✅ Found RPC call to replace');
  
  // Find and replace the entire RPC block
  const rpcBlockStart = content.indexOf(oldRpcCode);
  const firstSuccessCheck = content.indexOf('if (rpcError)', rpcBlockStart);
  const lastSuccessCheck = content.indexOf('devLogger.info(\'✅ Company and user_companies created successfully:', rpcBlockStart);
  const blockEnd = content.indexOf(');', lastSuccessCheck) + 2;
  
  const replacement = `const { success, companyId: createdCompanyId, error: createError } = await createCompanyDirectly(user.id, companyData);

        if (!success || !createdCompanyId) {
          throw new Error(createError || 'Failed to create company');
        }

        devLogger.info('✅ Company and user_companies created successfully:', createdCompanyId)`;
  
  content = content.slice(0, rpcBlockStart) + replacement + content.slice(blockEnd);
  console.log('✅ RPC call replaced with direct creation');
} else {
  console.log('⚠️  RPC call not found - may already be patched');
}

// Write back
fs.writeFileSync(filePath, content, 'utf8');
console.log('✅ File updated successfully');

File: patch_onboarding.py
#!/usr/bin/env python3
"""
Script pour corriger OnboardingContextNew.tsx
Remplace l'appel RPC create_company_with_user par create_company_with_setup
"""

file_path = 'src/contexts/OnboardingContextNew.tsx'

# Read the file
with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Check if already patched
if 'create_company_with_setup' in content and 'company_name_param' in content:
    print('✅ File already patched or using create_company_with_setup')
    exit(0)

# Try both apostrophe types
has_rpc_call = False
rpc_pattern = None

# Check for regular apostrophes first
if "await authedClient.rpc('create_company_with_user'" in content:
    print('✅ Found RPC call with regular apostrophes')
    has_rpc_call = True
    rpc_pattern = "await authedClient.rpc('create_company_with_user'"
# Check for curly apostrophes
elif "await authedClient.rpc('create_company_with_user'" in content:
    print('✅ Found RPC call with curly apostrophes')
    has_rpc_call = True
    rpc_pattern = "await authedClient.rpc('create_company_with_user'"

if has_rpc_call and rpc_pattern:
    # Find the start
    start_idx = content.find("const { data: result, error: rpcError } = ")
    
    if start_idx >= 0:
        print(f'✅ Found start at index {start_idx}')
        
        # Find the end: the last devLogger.info in this section about Company and user_companies created successfully
        # Search for the pattern within the context
        search_from = start_idx + 50
        
        # Look for devLogger.info('✅ Company and user_companies created successfully:
        success_idx = content.find("Company and user_companies created successfully:", search_from)
        
        if success_idx >= 0:
            print(f'✅ Found success message at index {success_idx}')
            end_idx = content.find(");", success_idx) + 2
            
            replacement = """const { data: newCompanyId, error: createError } = await supabase.rpc(
          'create_company_with_setup',
          {
            company_name_param: companyData.name,
            user_uuid_param: user.id,
            country_code_param: companyData.country || 'FR',
            currency_code_param: companyData.default_currency || 'EUR',
            accounting_standard_param: companyData.accounting_standard || null
          }
        );

        if (createError) {
          devLogger.error('❌ create_company_with_setup error:', createError);
          throw new Error(`Failed to create company: ${createError.message}`);
        }

        if (!newCompanyId) {
          throw new Error('No company ID returned from create_company_with_setup');
        }

        devLogger.info('✅ Company created successfully:', newCompanyId);"""
            
            # Replace
            new_content = content[:start_idx] + replacement + content[end_idx:]
            
            # Write back
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(new_content)
            
            print('✅ File patched successfully!')
            print('   - Replaced RPC call to create_company_with_user')
            print('   - Using create_company_with_setup instead')
        else:
            print('❌ Could not find success message')
            exit(1)
    else:
        print('❌ Could not find start of RPC block')
        exit(1)
else:
    print('⚠️  RPC call not found')
    # Try to find what's actually there
    if "await authedClient" in content:
        print('   Note: Found authedClient, RPC call might have different format')
    exit(0)
